src/Uno.UI.RemoteControl.Host/RemoteControlServer.cs

```diff
@@ -12,55 +12,103 @@
+		private readonly static Dictionary<string, AssemblyLoadContext> _loadContexts = new();
 
 		public RemoteControlServer(IConfiguration configuration)
 		{
 			_configuration = configuration;
-			_loadContext = new AssemblyLoadContext(null, isCollectible: true);
-			_loadContext.Unloading += (e) =>
+
+			if (this.Log().IsEnabled(LogLevel.Debug))
+			{
+				this.Log().LogDebug("Starting RemoteControlServer");
+			}
+		}
+
+		string IRemoteControlServer.GetServerConfiguration(string key)
+			=> _configuration[key] ?? "";
+
+		private void EnsureLoadContextForApplicationId(string applicationId)
+		{
+			if (_loadContexts.ContainsKey(applicationId))
+			{
+				return;
+			}
+
+			var loadContext = new AssemblyLoadContext(null, isCollectible: true);
+			loadContext.Unloading += (e) =>
 			{
 				if (this.Log().IsEnabled(LogLevel.Debug))
 				{
 					this.Log().LogDebug("Unloading assembly context");
 				}
 			};
-			_loadContext.Resolving += (context, assemblyName) =>
+
+			// Add custom resolving so we can find dependencies even when the processor assembly
+			// is built for a different .net version than the host process.
+			loadContext.Resolving += (context, assemblyName) =>
 			{
 				if (_loopAssembly is not null)
 				{
-					return context.LoadFromAssemblyPath(Path.Combine(Path.GetDirectoryName(_loopAssembly.Location)!, assemblyName.Name + ".dll"));
-				}
-				else
-				{
-					return context.LoadFromAssemblyName(assemblyName);
+					try
+					{
+						var loc = _loopAssembly.Location;
+						if (!string.IsNullOrWhiteSpace(loc))
+						{
+							var dir = Path.GetDirectoryName(loc);
+							if (!string.IsNullOrEmpty(dir))
+							{
+								var relPath = Path.Combine(dir, assemblyName.Name + ".dll");
+								if (File.Exists(relPath))
+								{
+									return context.LoadFromAssemblyPath(relPath);
+								}
+							}
+						}
+						else
+						{
+							if (this.Log().IsEnabled(LogLevel.Debug))
+							{
+								this.Log().LogDebug("Failed for identify location of dependency: {assemblyName}", assemblyName);
+							}
+						}
+					}
+					catch (Exception exc)
+					{
+						if (this.Log().IsEnabled(LogLevel.Error))
+						{
+							this.Log().LogError(exc, "Failed for load dependency: {assemblyName}", assemblyName);
+						}
+					}
 				}
+				return context.LoadFromAssemblyName(assemblyName);
 			};
 
-			if (this.Log().IsEnabled(LogLevel.Debug))
-			{
-				this.Log().LogDebug("Starting RemoteControlServer");
-			}
+			_loadContexts.Add(applicationId, loadContext);
 		}
 
-		string IRemoteControlServer.GetServerConfiguration(string key)
-			=> _configuration[key] ?? "";
-
-		private void RegisterProcessor(IServerProcessor hotReloadProcessor)
+		private AssemblyLoadContext GetAssemblyLoadContext(string applicationId)
 		{
-			_processors[hotReloadProcessor.Scope] = hotReloadProcessor;
+			EnsureLoadContextForApplicationId(applicationId);
+
+			return _loadContexts[applicationId];
```

Comment:
Prevent multiple lookups on _loadContexts by returning the context from EnsureLoadContextForApplicaitonId